types/json_binary_functions.go

```diff
@@ -244,7 +244,7 @@ func (bj BinaryJSON) Extract(pathExprList []JSONPathExpression) (ret BinaryJSON,
 		found = true
 		ret = buf[0]
 		// Fix https://github.com/pingcap/tidb/issues/30352
-		if pathExprList[0].ContainsAnyAsterisk() {
+		if pathExprList[0].ContainsAnyAsterisk() || pathExprList[0].ContainsAnyRange() {
```

Comment:
Maybe `ContainsAnySelection` instead of two judge?

Reply:
I replace them with `CouldMatchMultipleValues` :thinking: .